### ISSUE TITLE
Audio frames streaming and OPUS packets

### ISSUE THREAD

**Issue by JustNello** [livekit/python-sdks#96]
Hello,
thanks for this project :)

I'd like to transcribe an audio track with Deepgram, but I have some issues.

**The application**
The client is made of LiveKit React Components (ie. _LiveKitRoom_ and _AudioConference_), with REDundant encoding disabled when a client joins a room, as described in the [docs](https://docs.livekit.io/guides/audio-red/#disabling-red-when-a-client-joins-a-room).

The server uses this Python SDK and it has been implemented starting from the [Whisper](https://github.com/livekit/python-sdks/blob/main/examples/whisper/whisper.py) example. In my case, the "whisper_task" has been replaced by a "deepgram_task", in [this](https://gist.github.com/JustNello/b4d8c00146a3e440aabc39ebdef12590) gist.

**Issue**
I think I'm not getting how the AudioFrame (from rtc.AudioFrame package) encodes data. I'm new to audio streaming at all and this may be the cause of the issue. I know that the audio format is OPUS, but:

- Is either a containerized audio stream or raw audio stream? I refer to [this](https://github.com/livekit/python-sdks/blob/main/examples/whisper/whisper.py)
- What is the [frame duration](https://datatracker.ietf.org/doc/html/rfc6716#section-2.1.4)? Deepgram requires that the streaming buffer sizes should be between 20 milliseconds and 250 milliseconds of audio

In other words, what does `bytes(frame.data)` return? Is the OPUS packet?
I'm not able to inspect the packet using a [packet inspector](https://rameshvarun.github.io/binary-inspector/opus/).

Thank you in advance for any help you may give,
Luca

**Comment by theomonnom** [comment]
Hey Luca!
The frames you receive from the AudioStream are raw signed PCM.
Looking at the docs from [Deepgram](https://developers.deepgram.com/docs/encoding), they do support linear16.
I've already used Deepgram before, I think you can just connect to a Websocket and send the frames you receive from livekit directly.

**Comment by JustNello** [comment]
Awesome, it works 😀

One last question to improve my understanding: `rtc.RemoteTrackPublication.mime_type` yields _audio/opus_. When is the audio converted to signed PCM?

**Comment by theomonnom** [comment]
The mime_type is the codec being used while sending the media to the other end. libwebrtc will directly decode it once it receives it
